src/main/java/de/hysky/skyblocker/events/ChatEvents.java

```diff
@@ -0,0 +1,39 @@
+package de.hysky.skyblocker.events;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.text.Text;
+
+@Environment(EnvType.CLIENT)
+public class ChatEvents {
+	/**
+	 * This will be called when a game message is received, cancelled or not.
+	 */
+	public static final Event<ChatTextEvent> RECEIVE_TEXT = EventFactory.createArrayBacked(ChatTextEvent.class, listeners -> message -> {
+		for (ChatTextEvent listener : listeners) {
+			listener.onMessage(message);
+		}
+	});
+
+	/**
+	 * This will be called when a game message is received, cancelled or not.
+	 * This method is called with the result of {@link Text#getString()} to avoid each listener having to call it.
+	 */
+	public static final Event<ChatStringEvent> RECEIVE_STRING = EventFactory.createArrayBacked(ChatStringEvent.class, listeners -> message -> {
+		for (ChatStringEvent listener : listeners) {
+			listener.onMessage(message);
+		}
+	});
+
+	@FunctionalInterface
+	public interface ChatTextEvent {
+		void onMessage(Text message);
+	}
+
+	@FunctionalInterface
+	public interface ChatStringEvent {
+		void onMessage(String message);
```

Comment:
Shouldn't the overlay boolean be passed so that users can ignore action bar updates? It would be useful for ignoring action bar updates in the powder tracker unless powder messages show there too (I don't remember).

Reply:
They don't show up on the action bar afaik, but these events aren't fired when `overlay` is true
https://github.com/Emirlol/Skyblocker/blob/3d6642b5d90c583a89f91a05051a4794c7f8eeda/src/main/java/de/hysky/skyblocker/mixins/MessageHandlerMixin.java#L15
I didn't see a need for action bar message events so I left them as is for now, they should be fairly simple to add if anybody needs them.